exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -650,8 +651,15 @@ static Collection<? extends PointData> getPoints(MetricData metricData) {
     return Collections.emptyList();
   }
 
-  private static String metricName(String rawMetricName, PrometheusType type) {
-    String name = NameSanitizer.INSTANCE.apply(rawMetricName);
+  private static String metricName(MetricData rawMetric, PrometheusType type) {
+    String name = NameSanitizer.INSTANCE.apply(rawMetric.getName());
```

Comment:
If you move this line below the next `if` you can probably get rid of the `cleanUpString` method in `PrometheusUnitsHelper` -- no need to perform sanitization twice after all. (And the performance will benefit from the cache in `NameSanitizer` as well)